catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java

```diff
@@ -618,12 +618,20 @@ public Table deleteCustomMetric(UUID tableId, String columnName, String metricNa
   @Transaction
   public Table addDataModel(UUID tableId, DataModel dataModel) throws IOException {
     Table table = dao.findEntityById(tableId);
+    EntityReference oldOwner = table.getOwner();
     table.withDataModel(dataModel);
 
     // Carry forward the table description from the model to table entity, if empty
     if (nullOrEmpty(table.getDescription())) {
       table.setDescription(dataModel.getDescription());
     }
+
+    // Carry forward the table owner from the model to table entity, if empty
+    if (table.getOwner() == null) {
+      table.setOwner(dataModel.getOwner());
+      updateOwner(table, oldOwner, table.getOwner());
+    }
```

Comment:
```suggestion
    if (table.getOwner() == null) {
      storeOwner(table, dataModel.getOwner());
    }
```
And we can remove line 621.

Reply:
Yes making this change thanks 👍